src/Aspire.Hosting.Azure.Kusto/AzureKustoBuilderExtensions.cs

```diff
@@ -0,0 +1,218 @@
+    /// <param name="builder">The Kusto server resource builder.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="databaseName">The name of the database. If not provided, this defaults to the same value as <paramref name="name"/>.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<AzureKustoDatabaseResource> AddDatabase(this IResourceBuilder<AzureKustoClusterResource> builder, [ResourceName] string name, string? databaseName = null)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentException.ThrowIfNullOrWhiteSpace(name);
+
+        // Use the resource name as the database name if it's not provided
+        databaseName ??= name;
+
+        builder.Resource.AddDatabase(name, databaseName);
+        var kustoDatabase = new AzureKustoDatabaseResource(name, databaseName, builder.Resource);
+        var resourceBuilder = builder.ApplicationBuilder.AddResource(kustoDatabase);
+
+        // Register a health check that will be used to verify database is available
+        KustoConnectionStringBuilder? kcsb = null;
+        resourceBuilder.OnConnectionStringAvailable(async (db, evt, ct) =>
+        {
+            var connectionString = await db.ConnectionStringExpression.GetValueAsync(ct).ConfigureAwait(false) ??
+            throw new DistributedApplicationException($"ConnectionStringAvailableEvent published for resource '{db.Name}', but the connection string was null.");
+
+            kcsb = new KustoConnectionStringBuilder(connectionString);
+        });
+
+        var healthCheckKey = $"{kustoDatabase.Name}_check";
+        resourceBuilder.ApplicationBuilder
+            .Services
+            .AddHealthChecks()
+            .AddAzureKustoHealthCheck(healthCheckKey, _ => kcsb!);
+
+        return resourceBuilder
+            .WithHealthCheck(healthCheckKey);
+    }
+
+    /// <summary>
+    /// Configures the Kusto resource to run as an emulator using the Kustainer container.
+    /// </summary>
+    /// <remarks>
+    /// This version of the package defaults to the <inheritdoc cref="AzureKustoEmulatorContainerImageTags.Tag"/> tag of the <inheritdoc cref="AzureKustoEmulatorContainerImageTags.Registry"/>/<inheritdoc cref="AzureKustoEmulatorContainerImageTags.Image"/> container image.
+    /// </remarks>
+    /// <param name="builder">The resource builder to configure.</param>
+    /// <param name="httpPort">The host port that the Kusto HTTP endpoint is bound to.</param>
+    /// <param name="configureContainer">
+    /// Optional action to configure the Kusto emulator container.
+    /// </param>
+    /// <returns>The resource builder.</returns>
+    public static IResourceBuilder<AzureKustoClusterResource> RunAsEmulator(
+        this IResourceBuilder<AzureKustoClusterResource> builder,
+        int? httpPort = null,
```

Comment:
Remove this parameter. For `RunAsEmulator(...)` the convention is generally that that we only have the one argument `configureContainer`. If you want to provide a quick and easy way to bind to a http port you can implement a `WithHttpPort(...)` extension method (we have a few other examples in there).

Reply:
Removed!

I went without a `WithHttpPort` extension for now. I'm not sure if it'll actually be useful, and it's easier to add if necessary than to deprecate. I did however update the test to verify that the regular `WithEndpoint` call will work as expected.